main/actions.js

```diff
@@ -85,8 +85,38 @@ const listResults = () => {
   })
 }
 
+const showMeasurement = (msmtID) => {
+  const ooni = new Ooniprobe()
+  let measurement = {}
+
+  return new Promise((resolve, reject) => {
+    ooni.on('data', (data) => {
+      if (data.level == 'error') {
+        reject(data.message)
```

Comment:
How does `reject` work? Does it `throw`? Shouldn't you `return` here?

Reply:
`reject` doesn't 'throw'. It just passes control to the `catch` block of the promise handler. Not `catch`ing will throw an `Unhandled rejection exception` though.

And you are right. It is better to `return` here. Although not necessarily a problem.

Reply:
It's probably then an opportunity to apply this pattern to the other commands as well which don't.

As you said it should not however be a problem as you can only make a `Promise` reach it's final state (either `reject` or `resolve`) and any further call to `reject` or `resolve` shall result in a noop.

It's still probably a good idea to not waste time on those extra operations below in case of error and it's wise to fail early.

Reply:
Done